### DISCUSSION TITLE
All-pairs testing (pairwise testing)

### DISCUSSION THREAD

**Discussion by josephbuchma** [franiglesias/golden#12] [Ideas]
I'm quite new to this, and for some reason this technique does not seem to be very popular. Are you familiar with [all-pairs testing](https://en.wikipedia.org/wiki/All-pairs_testing) @franiglesias? Would be cool if `golden` package provided some utility for it.

**Comment by franiglesias** [comment]
I'm not familiar with this technique, but it is definitively an interesting topic. It seems kind of a fine-grained combinatory test. The original intent in golden is to reach the highest coverage in the least time, but the all-pairs technique seems to enable other goals, like better detection of bugs and testing optimization. 

I will need to seek for more information to get better understanding on how to choose the testing values for the all-pairs.
